docs/releases/2.0.txt

```diff
@@ -41,7 +41,12 @@ Minor features
 :mod:`django.contrib.admin`
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* ...
+* ``ModelAdmin.lookup_allowed`` now accepts a ``request`` parameter to ensure
+  it respects ``get_list_filter``. The parameter is optional to maintain
+  backwards compatibility.
+* A new ``ModelAdmin.get_changelist_instance`` method creates new override
```

Comment:
Other than using this in the new test, I don't see how it's linked to the rest of the PR. I think it should have its own ticket, tests, docs, and PR. That'll help simplify this one.